go-controller/pkg/ovn/base_secondary_layer2_network_controller.go

```diff
@@ -109,11 +121,33 @@ func (h *baseSecondaryLayer2NetworkControllerEventHandler) AddResource(obj inter
 func (h *baseSecondaryLayer2NetworkControllerEventHandler) UpdateResource(oldObj, newObj interface{}, inRetryCache bool) error {
 	switch h.objType {
 	case factory.NodeType:
-		node, ok := newObj.(*corev1.Node)
+		newNode, ok := newObj.(*corev1.Node)
 		if !ok {
 			return fmt.Errorf("could not cast %T object to Node", newObj)
 		}
-		return h.oc.addUpdateNodeEvent(node)
+		oldNode, ok := oldObj.(*kapi.Node)
+		if !ok {
+			return fmt.Errorf("could not cast oldObj of type %T to *kapi.Node", oldObj)
+		}
+		newNodeIsLocalZoneNode := h.oc.isLocalZoneNode(newNode)
+		nodeSubnetChanged := nodeSubnetChanged(oldNode, newNode)
+		if newNodeIsLocalZoneNode {
+			var nodeSyncsParam *nodeSyncs
+			if h.oc.isLocalZoneNode(oldNode) {
+				// determine what actually changed in this update
+				syncMgmtPort := macAddressChanged(oldNode, newNode) || nodeSubnetChanged
+				nodeSyncsParam = &nodeSyncs{syncMgmtPort: syncMgmtPort}
+			} else {
+				klog.Infof("Node %s moved from the remote zone %s to local zone %s.",
+					newNode.Name, util.GetNodeZone(oldNode), util.GetNodeZone(newNode))
+				// The node is now a local zone node. Trigger a full node sync.
+				nodeSyncsParam = &nodeSyncs{syncMgmtPort: true}
+			}
+
+			return h.oc.addUpdateLocalNodeEvent(newNode, nodeSyncsParam)
+		} else {
+			return h.oc.addUpdateRemoteNodeEvent(newNode)
+		}
```

Comment:
ditto, this is going to run for localnet too.

Reply:
Same here, as above, `addUpdateLocalNodeEvent()` doesn't do much in the localnet case.